src/ring/logger/messages.clj

```diff
@@ -43,17 +46,20 @@
 (defmulti request-details get-printer)
 
 (defmethod request-details :default
-  [{:keys [logger] :as options} req]
-  (debug logger (str "Request details: " (select-keys req [:character-encoding
-                                                    :content-length
-                                                    :content-type
-                                                    :query-string
-                                                    :remote-addr
-                                                    :request-method
-                                                    :scheme
-                                                    :server-name
-                                                    :server-port
-                                                    :uri]))))
+  [{:keys [logger log-query-string?] :as options} req]
+  (let [default-keys [:character-encoding
+                     :content-length
+                     :content-type
+                     :query-string
+                     :remote-addr
+                     :request-method
+                     :scheme
+                     :server-name
+                     :server-port
+                     :uri]]
+  (debug logger (str "Request details: " (select-keys req (if log-query-string? 
+                                                            default-keys
```

Comment:
Another minor thing:

I think it would read slightly better to remove query-string from `default-keys` and then invert this conditional to `conj default-keys :query-string`

But I'm fine if you prefer to keep it this way :)

Reply:
Also, I think the `debug` is missing some indentation, it seems to be at the same level as the `let` that wraps it

Reply:
Actually I've decided to replace it with `cond->`, seems to me both more flexible and readable, I hope you are ok with that.

Reply:
I also like the `cond->` better, thanks!